astropy/time/core.py

```diff
@@ -1269,13 +1310,13 @@ def __sub__(self, other):
         # we need a constant scale to calculate, which is guaranteed for
         # TimeDelta, but not for Time (which can be UTC)
         if other_is_delta:  # T - Tdelta
+            out = self.replicate()
             if self.scale in other.SCALES:
-                out = self.replicate()
                 if other.scale not in (out.scale, None):
                     other = getattr(other, out.scale)
             else:
-                out = getattr(self, (other.scale if other.scale is not None
-                                     else 'tai'))
+                out = getattr(out, (other.scale if other.scale is not None
```

Comment:
This change (and the moving up of `out = self.replicate()`) has me confused: why is it necessary, given that changing scale already implies a replicate?
```
    def __getattr__(self, attr):
        if attr in self.SCALES and self.scale is not None:
            tm = self.replicate()
            tm._set_scale(attr)
            return tm
```